bgpd/bgpd.c

```diff
@@ -7509,6 +7509,39 @@ int peer_ttl_security_hops_unset(struct peer *peer)
 	return ret;
 }
 
+static void peer_reset_message_stats(struct peer *peer)
+{
+	time_t now;
+
+	if (peer) {
+		time(&now);
+		atomic_store_explicit(&peer->update_time, now,
```

Comment:
update_time is when we received the last update packet from a peer.  I do not think this should be changed here.  Nor is this an atomic operation?

Reply:
I thought setting it to the current epoch would be equivalent to setting it to 0, since the time difference will be derived from this value and the epoch at that time.

The value is defined as `_Atomic` in `bgpd/bgpd.h`:
```
	_Atomic time_t update_time;       /* Update message received time. */
```

I will take that part out of the code. In retrospective, as you said, it does not make much sense to reset it.

Reply:
Timer reset is removed in latest push.